### ISSUE TITLE
References are not resolved relative to the referring document

### ISSUE THREAD

**Issue by mkauf** [Mermade/swagger2openapi#39]
### Checklist

- [x] Conversion: I have checked my source definition is valid OpenAPI 2.0

### Detailed Description

I have split a Swagger specification into multiple files. ``swagger2openapi`` cannot resolve the file "ok.yaml" because it tries to read "subdir2/ok.yaml" instead of "subdir1/subdir2/ok.yaml".

```
$ swagger2openapi -r index.yaml 
{ Error: ENOENT: no such file or directory, open '/./subdir2/ok.yaml'
    at Error (native)
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/./subdir2/ok.yaml' }
```

index.yaml:
```
swagger: '2.0'
info:
  version: 0.0.0
  title: Simple API
paths:
  /test:
    $ref: ./subdir1/test.yaml
```

subdir1/test.yaml:
```
get:
  responses:
    200:
      description: OK
      schema:
        $ref: ./subdir2/ok.yaml
```

subdir1/subdir2/ok.yaml:
```
type: object
properties:
  name:
    type: string
```

**Comment by MikeRalphson** [comment]
Many thanks for reporting this, and the detailed issue description!

Are you by any chance able to test the fix from GitHub? It's in the `issue39` branch, this commit: https://github.com/Mermade/swagger2openapi/commit/2851f1ecfc218085aa8573e9d1a2b4f9bf3ec2d0 I've tested it with your example and `http` resolution and it looks ok, but I'd appreciate confirmation.

**Comment by mkauf** [comment]
Thanks for your quick response!

It does not work yet if I change the reference in "subdir1/test.yaml" to a reference with a protocol, for example:
```
$ref: http://myserver/subdir1/subdir2/ok.yaml
```

I propose this change, I have tested it and it works for me:
```
diff --git a/common.js b/common.js
index d53e7c5..c2aebcc 100644
--- a/common.js
+++ b/common.js
@@ -71,9 +71,8 @@ function resolveExternal(root, pointer, options, callback) {
 
     let u2 = url.parse(pointer);
     let effectiveProtocol = (u2.protocol ? u2.protocol : (u.protocol ? u.protocol : 'file:'));
-    if (u2.protocol) pointer = u2.path;
 
-    let target = (base ? base+'/': '')+pointer;
+    let target = url.resolve(base ? base + '/' : '', pointer);
 
     if (options.cache[target]) {
         if (options.verbose) console.log('CACHED',target);
```

**Comment by MikeRalphson** [comment]
Should be fixed in swagger2openapi@2.11.3

**Comment by mkauf** [comment]
The new version works great. Thanks!
